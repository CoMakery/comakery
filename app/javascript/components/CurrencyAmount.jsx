import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  color: #201662;
`

const Amount = styled.div`
  margin-right: 0.5em;
`

const Currency = styled.div`
  margin-right: 0.5em;
  text-transform: uppercase;
`

const Logo = styled.img`
  height: 14px;
`

class CurrencyAmount extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Wrapper>
          <Amount>
            {this.props.amount}
          </Amount>
          <Currency>
            {this.props.currency || '(No Token)'}
          </Currency>
          {this.props.logoUrl &&
            <Logo src={this.props.logoUrl} alt={`${this.props.currency} logo`} />
          }
        </Wrapper>
      </React.Fragment>
    )
  }
}

CurrencyAmount.propTypes = {
  amount  : PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  logoUrl : PropTypes.string.isRequired
}
CurrencyAmount.defaultProps = {
  amount  : '',
  currency: '',
  logoUrl : ''
}
export default CurrencyAmount
