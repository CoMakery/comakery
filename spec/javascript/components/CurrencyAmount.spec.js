import React from 'react'
import { mount } from 'enzyme'
import CurrencyAmount from 'components/CurrencyAmount'

describe('CurrencyAmount', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<CurrencyAmount />)

    expect(wrapper).toMatchSnapshot()
  })
})
