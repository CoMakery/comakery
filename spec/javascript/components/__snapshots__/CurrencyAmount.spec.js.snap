// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyAmount renders component correctly 1`] = `
<CurrencyAmount
  amount=""
  currency=""
  logoUrl=""
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "khtMgg",
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
  font-family: Montserrat;
  font-size: 14px;
  font-weight: bold;
  color: #201662;
  padding-right: 15px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa khtMgg"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "eNKWhX",
                  "rules": Array [
                    "
  margin-right: 0.5em;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH eNKWhX"
            />
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "eglqcF",
                  "rules": Array [
                    "
  margin-right: 0.5em;
  text-transform: uppercase;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat eglqcF"
            >
              (No Token)
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</CurrencyAmount>
`;
